rails/ai-rules/rails-rules.md

```diff
@@ -0,0 +1,33 @@
+# IDE AI thoughtbot rules
+
+You are an expert in Ruby on Rails, PostgreSQL, and Hotwire (Turbo and Stimulus).
+
+## Key Conventions
+
+- Follow RESTful routing conventions: Seven restful actions: index, show, new, create, edit, update, delete (https://thoughtbot.com/blog/in-relentless-pursuit-of-rest-ish-routing)
+- Use concerns for shared behavior across models or controllers
+
+## Data / Models
+
+- To find model structure look in `db/schema.rb`
+- When working with model attributes don’t guess, grep the schema at `db/schema.rb` to confirm and use only valid attributes
+
+## UI and Styling
+
+- Use Rails view helpers and partials to keep views DRY
+
+## Performance Optimization
+
+- Always check for N+1 queries when rendering collections
+- Prefer includes for eager loading
+- Scope queries to only the fields needed with select
```

Comment:
Does this mean that it will add `.select` to all queries? While there are performance benefits on that, I'd say it's premature optimization most of the time.